internal/storage/storageutil/client_helper.go

```diff
@@ -0,0 +1,110 @@
+package storageutil
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/auth"
+	mountpkg "github.com/googlecloudplatform/gcsfuse/internal/mount"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+)
+
+const CustomEndpoint = "https://localhost:9000"
+const DummyKeyFile = "test/test_creds.json"
+
+type StorageClientConfig struct {
+	ClientProtocol      mountpkg.ClientProtocol
+	MaxConnsPerHost     int
+	MaxIdleConnsPerHost int
+	HttpClientTimeout   time.Duration
+	MaxRetryDuration    time.Duration
+	RetryMultiplier     float64
+	UserAgent           string
+	Endpoint            *url.URL
+	KeyFile             string
+	TokenUrl            string
+	ReuseTokenFromUrl   bool
+}
+
+// GetDefaultStorageClientConfig is only for test, making the default endpoint
+// non-nil, so that we can create dummy tokenSource while unit test.
+func GetDefaultStorageClientConfig() (clientConfig StorageClientConfig) {
```

Comment:
dont put test and non-test code in the same file.

Reply:
Moving this method to test-file.

Reply:
Ohh, we can't do, since this is used by test of storage package.

Reply:
You can't put test and non-test code in the same file.

Reply:
Added a new file (test.go) containing test related method.